### PR TITLE
Enable run-script for running latest rust-optimizer version

### PR DESCRIPTION
**Pull request by orkunkl**
After the PR:
```shell
cargo run-script optimize
```
Will run:
```shell
docker run --rm -v "$(pwd)":/code \
  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
  cosmwasm/rust-optimizer:0.12.0
```

### CODE REVIEW

**Comment by ethanfrey** [comment]
Interesting find: https://github.com/JoshMcguigan/cargo-run-script

Why not use a normal Makefile or shell script? I guess as people must have Rust installed already?

There is also [cargo-make](https://github.com/sagiegurari/cargo-make) which is interesting but probably way too powerful (but might be interesting if you add more scripts)

Anyway, looks good to me, maybe you look at other tools if you add more scripts.

**Review by ethanfrey — APPROVED** [review]
lgtm

File: Cargo.toml [hunk 1]
```diff
@@ -32,6 +32,13 @@ backtraces = ["cosmwasm-std/backtraces"]
 # use library feature to disable all instantiate/execute/query exports
 library = []
 
+[package.metadata.scripts]
+optimize = """docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.12.0
```
Comment (webmaster128) [inline Cargo.toml:39]:
```suggestion
  cosmwasm/rust-optimizer:0.12.3
```

File: README.md [hunk 2]
```diff
@@ -13,12 +13,12 @@ Assuming you have a recent version of rust and cargo (v1.51.0+) installed
 (via [rustup](https://rustup.rs/)),
 then the following should get you a new repo to start a contract:
 
-First, install
-[cargo-generate](https://github.com/ashleygwilliams/cargo-generate).
+
+Install [cargo-generate](https://github.com/ashleygwilliams/cargo-generate) and cargo-run-script.
```
Comment (webmaster128) [inline README.md:17]: Hmm, can't we use a native way to do thart? Like the aliases defined in `.cargo/config`

**Review by webmaster128 — COMMENTED** [review]
Interesting, but not a fan of adding some random external plugin here

File: README.md (hunk 2 above)
Comment (ethanfrey) [reply README.md:17]: Aliases only let you use cargo subcommands, not call shell scripts.

Adding `optimize.sh` or a Makefile are two alternative ways that depend on a more common tool (bash / make). But we do need a tool

Comment (ethanfrey) [reply README.md:17]: (I have tried to do more with cargo aliases before)

Comment (webmaster128) [reply README.md:17]: Ah, right, thanks

**Review by webmaster128 — APPROVED** [review]
Alright then. I have no better idea to solve this.

**Pull request merged**